internal/db/fetcher/indexer_matchers.go

```diff
@@ -55,16 +55,34 @@ func (m *intMatcher) Match(value client.NormalValue) (bool, error) {
 	return false, NewErrUnexpectedTypeValue[int64](value)
 }
 
-type floatMatcher struct {
+type float32Matcher struct {
+	value    float32
+	evalFunc func(float32, float32) bool
+}
+
+func (m *float32Matcher) Match(value client.NormalValue) (bool, error) {
```

Comment:
suggestion: The whole function is showing as untested, please add tests if not there already (maybe due to ci failures, ignore if tested already).